File: imperative/java-imperative/src/main/java/com/fernandoglizt/imperative/javaimperative/model/ProcessRequest.java
package com.fernandoglizt.imperative.javaimperative.model;

public record ProcessRequest(
    int count,
    int batch,
    int ioDelayMs,
    String downstreamUrl
) {
    public ProcessRequest {
        if (count <= 0) {
            throw new IllegalArgumentException("count must be > 0");
        }
        if (batch <= 0) {
            throw new IllegalArgumentException("batch must be > 0");
        }
    }

    public String getDownstreamUrl() {
        return downstreamUrl != null && !downstreamUrl.isBlank() 
            ? downstreamUrl 
            : "http://slow-io:8080/slow";
    }

    public int getIoDelayMs() {
        return Math.max(0, ioDelayMs);
    }
}

File: imperative/kotlin-imperative/src/main/kotlin/com/fernandoglizt/imperative/kotlinimperative/model/ProcessRequest.kt
package com.fernandoglizt.imperative.kotlinimperative.model

data class ProcessRequest(
    val count: Int,
    val batch: Int,
    val ioDelayMs: Int = 0,
    val downstreamUrl: String? = null
) {
    init {
        require(count > 0) { "count must be > 0" }
        require(batch > 0) { "batch must be > 0" }
    }

    fun getDownstreamUrlOrDefault(): String =
        downstreamUrl?.takeIf { it.isNotBlank() } ?: "http://slow-io:8080/slow"

    fun getIoDelayMsNormalized(): Int = maxOf(0, ioDelayMs)
}

